Error building on GitHub Actions + R 4.2

This looks like a openssl issue, but doesn't seem to be present in the CRAN checks on Windows:

https://github.com/r-spatial/s2/runs/6435327413?check_suite_focus=true#step:5:17596

```
  OE> g++ -std=gnu++11 -shared -s -static-libgcc -o s2.dll tmp.def RcppExports.o cpp-compat.o init.o s2-accessors.o s2-bounds.o s2-cell-union.o s2-cell.o s2-constructors-formatters.o s2-geography.o s2-lnglat.o s2-matrix.o s2-point.o s2-predicates.o s2-transformers.o s2-xptr.o tessellator.o wk-c-utils.o wk-impl.o -Ls2 -ls2static -L../windows/openssl-1.1.1k/lib/x64-ucrt -lssl -lcrypto -lcrypt32 -lws2_32 -Lc:/rtools42/x86_64-w64-mingw32.static.posix/lib/x64 -Lc:/rtools42/x86_64-w64-mingw32.static.posix/lib -LC:/R/bin/x64 -lR
  OE> C:/rtools40/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/8.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: ../windows/openssl-1.1.1k/lib/x64-ucrt/libcrypto.a(e_capi.o):(.text+0x3700): undefined reference to `__imp___acrt_iob_func'
  OE> C:/rtools40/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/8.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: ../windows/openssl-1.1.1k/lib/x64-ucrt/libcrypto.a(eng_openssl.o):(.text+0x12): undefined reference to `__imp___acrt_iob_func'
  OE> C:/rtools40/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/8.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: ../windows/openssl-1.1.1k/lib/x64-ucrt/libcrypto.a(eng_openssl.o):(.text+0x266): undefined reference to `__imp___acrt_iob_func'
  OE> C:/rtools40/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/8.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: ../windows/openssl-1.1.1k/lib/x64-ucrt/libcrypto.a(ui_openssl.o):(.text+0x19): undefined reference to `__imp___acrt_iob_func'
  OE> C:/rtools40/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/8.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: ../windows/openssl-1.1.1k/lib/x64-ucrt/libcrypto.a(ui_openssl.o):(.text+0x174): undefined reference to `__imp___acrt_iob_func'
  OE> C:/rtools40/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/8.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: ../windows/openssl-1.1.1k/lib/x64-ucrt/libcrypto.a(ui_openssl.o):(.text+0x197): more undefined references to `__imp___acrt_iob_func' follow
  OE> C:/rtools40/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/8.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: ../windows/openssl-1.1.1k/lib/x64-ucrt/libcrypto.a(v3_utl.o):(.text+0xd2a): undefined reference to `__stdio_common_vsscanf'
  OE> collect2.exe: error: ld returned 1 exit status
```

--- a/.github/workflows/check-standard.yaml
+++ b/.github/workflows/check-standard.yaml
@@ -20,7 +20,7 @@ jobs:
         config:
           - {os: macOS-latest,   r: 'release'}
           - {os: windows-latest, r: '3.6'}
-          - {os: windows-latest, r: 'release'}
+          - {os: windows-latest, r: 'release', ssl_backend: 'openssl'}
           - {os: ubuntu-latest,   r: 'devel', http-user-agent: 'release'}
           - {os: ubuntu-latest,   r: 'release'}
           - {os: ubuntu-latest,   r: 'oldrel-1'}
@@ -45,6 +45,8 @@ jobs:
           extra-packages: rcmdcheck
 
       - uses: r-lib/actions/check-r-package@v1
+        env:
+          CURL_SSL_BACKEND: ${{ matrix.config.ssl_backend }}
 
       - name: Show install output
         if: always()

--- a/.github/workflows/check-standard.yaml
+++ b/.github/workflows/check-standard.yaml
@@ -20,7 +20,7 @@ jobs:
         config:
           - {os: macOS-latest,   r: 'release'}
           - {os: windows-latest, r: '3.6'}
-          - {os: windows-latest, r: 'release', rtools-version: '42'}
+          - {os: windows-latest, r: 'release'}
           - {os: ubuntu-latest,   r: 'devel', http-user-agent: 'release'}
           - {os: ubuntu-latest,   r: 'release'}
           - {os: ubuntu-latest,   r: 'oldrel-1'}

--- a/.github/workflows/check-standard.yaml
+++ b/.github/workflows/check-standard.yaml
@@ -20,7 +20,7 @@ jobs:
         config:
           - {os: macOS-latest,   r: 'release'}
           - {os: windows-latest, r: '3.6'}
-          - {os: windows-latest, r: 'release', ssl_backend: 'openssl'}
+          - {os: windows-latest, r: 'release'}
           - {os: ubuntu-latest,   r: 'devel', http-user-agent: 'release'}
           - {os: ubuntu-latest,   r: 'release'}
           - {os: ubuntu-latest,   r: 'oldrel-1'}
@@ -34,19 +34,17 @@ jobs:
 
       - uses: r-lib/actions/setup-pandoc@v1
 
-      - uses: r-lib/actions/setup-r@v1
+      - uses: r-lib/actions/setup-r@v2
         with:
           r-version: ${{ matrix.config.r }}
           http-user-agent: ${{ matrix.config.http-user-agent }}
           use-public-rspm: true
 
-      - uses: r-lib/actions/setup-r-dependencies@v1
+      - uses: r-lib/actions/setup-r-dependencies@v2
         with:
           extra-packages: rcmdcheck
 
-      - uses: r-lib/actions/check-r-package@v1
-        env:
-          CURL_SSL_BACKEND: ${{ matrix.config.ssl_backend }}
+      - uses: r-lib/actions/check-r-package@v2
 
       - name: Show install output
         if: always()

--- a/configure
+++ b/configure
@@ -31,14 +31,14 @@ elif [ "$PKGCONFIG_CFLAGS" ] || [ "$PKGCONFIG_LIBS" ]; then
   PKG_CFLAGS=${PKGCONFIG_CFLAGS}
   PKG_LIBS=${PKGCONFIG_LIBS}
 elif [ `uname` = "Darwin" ]; then
-  brew --version 2>/dev/null
+  test ! "$CI" && brew --version 2>/dev/null
   if [ $? -eq 0 ]; then
-    BREWDIR=`brew --prefix openssl@1.1`
-    PKG_CFLAGS="-I$BREWDIR/include"
-    PKG_LIBS="-L$BREWDIR/lib $PKG_LIBS"
+    BREWDIR=`brew --prefix`
+    PKG_CFLAGS="-I$BREWDIR/opt/openssl/include -I$BREWDIR/opt/openssl@1.1/include"
+    PKG_LIBS="-L$BREWDIR/opt/openssl/lib -L$BREWDIR/opt/openssl@1.1/lib $PKG_LIBS"
   else
     curl -sfL "https://autobrew.github.io/scripts/$PKG_BREW_NAME" > autobrew
-    . autobrew
+    . ./autobrew
   fi
 fi
 
@@ -83,7 +83,7 @@ fi
 
 # Test if versioned linking works
 ${CC} ${PKG_CFLAGS} src/tests/main.c ${PKG_LIBS_VERSIONED} -o src/main.exe 2>/dev/null
-if [ $? -eq 0 ]; then PKG_LIBS="${PKG_LIBS_VERSIONED}"; rm src/main.exe; fi
+if [ $? -eq 0 ]; then PKG_LIBS="${PKG_LIBS_VERSIONED}"; fi
 
 # Suppress opensslv3 warnings for now
 if [ "$SONAME" = "3" ]; then

--- a/src/Makevars.win
+++ b/src/Makevars.win
@@ -1,5 +1,5 @@
 PKG_CPPFLAGS = -DS2_USE_EXACTFLOAT -D_USE_MATH_DEFINES -DNDEBUG -DIS_LITTLE_ENDIAN -DOMIT_STRPTIME -I../windows/openssl-1.1.1k/include -I../src
-PKG_LIBS = -Ls2 -ls2static -L../windows/openssl-1.1.1k/lib${R_ARCH}${CRT} -lssl -lcrypto -lcrypt32 -lws2_32
+PKG_LIBS = -Ls2 -ls2static -L../windows/openssl-1.1.1k/lib${R_ARCH}${CRT} -lssl -lcrypto -lws2_32 -lgdi32 -lcrypt32
 
 CXX_STD = CXX11
 

--- a/src/Makevars.win
+++ b/src/Makevars.win
@@ -1,5 +1,6 @@
-PKG_CPPFLAGS = -DS2_USE_EXACTFLOAT -D_USE_MATH_DEFINES -DNDEBUG -DIS_LITTLE_ENDIAN -DOMIT_STRPTIME -I../windows/openssl-1.1.1k/include -I../src
-PKG_LIBS = -Ls2 -ls2static -L../windows/openssl-1.1.1k/lib${R_ARCH}${CRT} -lssl -lcrypto -lws2_32 -lgdi32 -lcrypt32
+VERSION = 1.1.1k
+PKG_CPPFLAGS = -DS2_USE_EXACTFLOAT -D_USE_MATH_DEFINES -DNDEBUG -DIS_LITTLE_ENDIAN -DOMIT_STRPTIME -I../windows/openssl-$(VERSION)/include -I../src
+PKG_LIBS = -Ls2 -ls2static -L../windows/openssl-$(VERSION)/lib${R_ARCH}${CRT} -lssl -lcrypto -lws2_32 -lgdi32 -lcrypt32
 
 CXX_STD = CXX11
 
@@ -205,7 +206,7 @@ $(S2LIBS): winlibs
 
 winlibs:
 	mkdir -p ../inst
-	"${R_HOME}/bin${R_ARCH_BIN}/Rscript.exe" --vanilla "../tools/winlibs.R"
+	"${R_HOME}/bin${R_ARCH_BIN}/Rscript.exe" "../tools/winlibs.R" $(VERSION)
 
 clean:
 	rm -f $(SHLIB) $(STATLIB) $(OBJECTS) $(S2LIBS)

--- a/tools/winlibs.R
+++ b/tools/winlibs.R
@@ -1,7 +1,9 @@
-# Build against openssl libraries that were compiled with the Rtools gcc toolchain.
-if(!file.exists("../windows/openssl-1.1.1k/include/openssl/ssl.h")){
+# Build against mingw-w64 build of openssl
+VERSION <- commandArgs(TRUE)
+if(!file.exists(sprintf("../windows/openssl-%s/include/openssl/ssl.h", VERSION))){
   if(getRversion() < "3.3.0") setInternet2()
-  download.file("https://github.com/rwinlib/openssl/archive/v1.1.1k.zip", "lib.zip", quiet = TRUE)
+  download.file(sprintf("https://github.com/rwinlib/openssl/archive/v%s.zip", VERSION),
+                "lib.zip", quiet = TRUE)
   dir.create("../windows", showWarnings = FALSE)
   unzip("lib.zip", exdir = "../windows")
   unlink("lib.zip")

